src/main/java/org/terasology/launcher/LauncherInitTask.java

```diff
@@ -120,51 +127,34 @@ private OperatingSystem getOperatingSystem() throws LauncherStartFailedException
         return os;
     }
 
-    private Path getLauncherDirectory(OperatingSystem os) throws LauncherStartFailedException {
-        logger.trace("Init LauncherDirectory...");
-        final Path launcherDirectory = DirectoryUtils.getApplicationDirectory(os, DirectoryUtils.LAUNCHER_APPLICATION_DIR_NAME);
+    private void initDirectory(Path dir, String errorLabel, DirectoryCreator... creators)
```

Comment:
The whole set of methods dealing with directory creation, initialization, etc need to get a revisit. Cleaning up naming, adding Javadoc, refactoring (more) common code.